### PR TITLE
Converts some notify_ghosts args to bitflags, multilines all notify_ghosts calls

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

This helps clean up my favorite helper proc in the whole codebase, `notify_ghosts()`.

The notify_suiciders, ignore_mapload, and flashwindow args are GONE. They have been replaced with the notify_flags bitflag argument. This was intended to make deadchat announcements a bitflag argument too, but those got reverted right before I originally wanted to submit this PR.

The on-screen popup now shows the notification body when you hover it with your mouse again. The format is now `[notify_ghosts message] -- [click action (orbit/jump/play)]`

Every single `notify_ghosts()` call has been changed to multiline format and has been given trailing commas. Pretty!
## Why It's Good For The Game

Cleans up a proc that is very popular and going through a lot of changes at the time.

Allows for further flexibility when this proc inevitably gets tweaked or improved. 12 -> 10 args is an improvement, and it doesn't impact the helper's flexibility at all.
## Changelog
:cl: Rhials
code: The notify_ghosts proc has been cleaned up. Please report any abnormal changes in deadchat notification behavior.
qol: The on-screen deadchat popups now contain the notification blurb when hovered with your mouse again.
/:cl:

### CODE REVIEW

File: code/__DEFINES/observers.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#define GHOST_NOTIFY_IGNORE_MAPLOAD (1<<0)
+/// Determines if the notification will flash the Byond window.
+#define GHOST_NOTIFY_FLASH_WINDOW (1<<1)
+/// Determines if the notification will notify suiciders.
+#define GHOST_NOTIFY_NOTIFY_SUICIDERS (1<<2)
+
+/// The default set of flags to be passed into a notify_ghosts call.
+#define NOTIFY_CATEGORY_DEFAULT (GHOST_NOTIFY_FLASH_WINDOW | GHOST_NOTIFY_IGNORE_MAPLOAD | GHOST_NOTIFY_NOTIFY_SUICIDERS)
+/// The default set of flags, without the flash_window flag.
+#define NOTIFY_CATEGORY_NOFLASH (GHOST_NOTIFY_IGNORE_MAPLOAD | GHOST_NOTIFY_NOTIFY_SUICIDERS)
```
Comment (MrMelbert) [inline code/__DEFINES/observers.dm:12]:
```suggestion
#define NOTIFY_CATEGORY_NOFLASH (NOTIFY_CATEGORY_DEFAULT & ~GHOST_NOTIFY_FLASH_WINDOW)
```
(You don't *have* to do this but this way it means someone doesn't have to update both flags if they change the default)

File: code/datums/components/orbit_poll.dm [hunk 2]
```diff
@@ -47,13 +47,14 @@
 
 	var/message = custom_message || "[capitalize(src.title)] is looking for volunteers"
 
-	notify_ghosts("[message]. An orbiter will be chosen in [DisplayTimeText(timeout)].\n", \
+	notify_ghosts(
+		"[message]. An orbiter will be chosen in [DisplayTimeText(timeout)].\n", \
 		action = NOTIFY_ORBIT, \
 		enter_link = "<a href='?src=[REF(src)];ignore=[ignore_key]'>(Ignore)</a>", \
-		flashwindow = FALSE, \
+		notify_flags = NOTIFY_CATEGORY_DEFAULT, \
```
Comment (MrMelbert) [inline code/datums/components/orbit_poll.dm:54]: Is this supposed to be `NOTIFY_CATEGORY_NOFLASH`?

File: code/modules/food_and_drinks/machinery/microwave.dm [hunk 3]
```diff
@@ -578,7 +578,13 @@
 			non_food_ingedients--
 		if(istype(potential_fooditem, /obj/item/modular_computer/pda) && prob(75))
 			pda_failure = TRUE
-			notify_ghosts("[cooker] has overheated their PDA!", source = src, action = NOTIFY_JUMP, flashwindow = FALSE, header = "Hunger Games: Catching Fire")
+			notify_ghosts(
+				"[cooker] has overheated their PDA!",
+				source = src,
+				action = NOTIFY_JUMP,
```
Comment (jlsnow301) [inline code/modules/food_and_drinks/machinery/microwave.dm:584]: this is the default
```suggestion
```

File: code/datums/components/orbit_poll.dm [hunk 4]
```diff
@@ -47,13 +47,14 @@
...
+	notify_ghosts(
+		"[message]. An orbiter will be chosen in [DisplayTimeText(timeout)].\n", \
 		action = NOTIFY_ORBIT, \
 		enter_link = "<a href='?src=[REF(src)];ignore=[ignore_key]'>(Ignore)</a>", \
-		flashwindow = FALSE, \
+		notify_flags = NOTIFY_CATEGORY_NOFLASH, \
 		header = "Volunteers requested", \
 		ignore_key = ignore_key, \
-		source = parent \
+		source = parent, \
```
Comment (jlsnow301) [inline code/datums/components/orbit_poll.dm:57]: (this is my fault) can you please remove the unneeded slashes

File: code/modules/events/_event.dm [hunk 5]
```diff
@@ -229,7 +229,12 @@ Runs the event
 /datum/round_event/proc/announce_to_ghosts(atom/atom_of_interest)
 	if(control.alert_observers)
 		if (atom_of_interest)
-			notify_ghosts("[control.name] has an object of interest: [atom_of_interest]!", source=atom_of_interest, action=NOTIFY_ORBIT, header="Something's Interesting!")
+			notify_ghosts(
+				"[control.name] has an object of interest: [atom_of_interest]!",
+				source=atom_of_interest,
+				action=NOTIFY_ORBIT,
+				header="Something's Interesting!",
```
Comment (jlsnow301) [inline code/modules/events/_event.dm:236]:
```suggestion
				source = atom_of_interest,
				action = NOTIFY_ORBIT,
				header = "Something's Interesting!",
```

File: code/modules/mapfluff/ruins/objects_and_mobs/necropolis_gate.dm [hunk 6]
```diff
@@ -190,7 +190,13 @@ GLOBAL_DATUM(necropolis_gate, /obj/structure/necropolis_gate/legion_gate)
 				M.playsound_local(T, null, 100, FALSE, 0, FALSE, pressure_affected = FALSE, sound_to_use = legion_sound)
 				flash_color(M, flash_color = "#FF0000", flash_time = 50)
 		var/mutable_appearance/release_overlay = mutable_appearance('icons/effects/effects.dmi', "legiondoor")
-		notify_ghosts("Legion has been released in the [get_area(src)]!", source = src, alert_overlay = release_overlay, action = NOTIFY_JUMP, flashwindow = FALSE)
+		notify_ghosts(
+			"Legion has been released in the [get_area(src)]!",
+			source = src,
+			alert_overlay = release_overlay,
+			action = NOTIFY_JUMP,
```
Comment (jlsnow301) [inline code/modules/mapfluff/ruins/objects_and_mobs/necropolis_gate.dm:197]:
```suggestion
```

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 7]
```diff
@@ -672,9 +672,13 @@
 		return
 	to_chat(user, span_notice("You call out for aid, attempting to summon spirits to your side."))
 
-	notify_ghosts("[user] is raising [user.p_their()] [name], calling for your help!",
-		enter_link="<a href=?src=[REF(src)];orbit=1>(Click to help)</a>",
-		source = user, ignore_key = POLL_IGNORE_SPECTRAL_BLADE, header = "Spectral blade")
+	notify_ghosts(
+		"[user] is raising [user.p_their()] [name], calling for your help!",
+		enter_link = "<a href=?src=[REF(src)];orbit=1>(Click to help)</a>",
```
Comment (jlsnow301) [inline code/modules/mining/lavaland/megafauna_loot.dm:677]:
```suggestion
		action = NOTIFY_ORBIT,
```

File: code/modules/mob_spawn/ghost_roles/spider_roles.dm [hunk 8]
```diff
@@ -103,7 +103,18 @@
 	amount_grown += rand(5, 15) * seconds_per_tick
 	if(amount_grown >= 100 && !ready)
 		ready = TRUE
-		notify_ghosts("[src] is ready to hatch!", null, enter_link = "<a href=?src=[REF(src)];activate=1>(Click to play)</a>", source = src, action = NOTIFY_ORBIT, ignore_key = POLL_IGNORE_SPIDER, flashwindow = flash_window)
+		var/notify_flags_to_pass = NOTIFY_CATEGORY_NOFLASH
+		if(flash_window)
+			notify_flags_to_pass &= GHOST_NOTIFY_FLASH_WINDOW
+		notify_ghosts(
+			"[src] is ready to hatch!",
+			null,
```
Comment (jlsnow301) [inline code/modules/mob_spawn/ghost_roles/spider_roles.dm:111]:
```suggestion
```

File: code/modules/mob_spawn/ghost_roles/spider_roles.dm [hunk 9]
```diff
@@ -103,7 +103,18 @@
...
+			enter_link = "<a href=?src=[REF(src)];activate=1>(Click to play)</a>",
```
Comment (jlsnow301) [inline code/modules/mob_spawn/ghost_roles/spider_roles.dm:112]: duplicate of NOTIFY_PLAY
```suggestion
```

**Review by jlsnow301 — COMMENTED** [review]
just some nits. perhaps out of scope for the PR but in the future we could just merge the NOTIFY_ORBIT and NOTIFY_JUMP actions into a singular default that orbits the source unless its a turf. then we'd only need the argument for a few boolean cases where clicking it would interact/attack

**Pull request merged**